apps/dashboard/app/batch-payments/server-actions.ts

```diff
@@ -159,3 +181,82 @@ export const processPaymentsServerAction = async (
     responsePayload: null,
   }
 }
+
+export const validatePaymentDetail = async (
+  TotalAmount: TotalAmountForWallets,
+): Promise<{
+  error: boolean
+  message: string
+  responsePayload: {
+    transactionAmount: {
+      totalAmountForBTCWallet: number
+      totalAmountForUSDWallet: number
+    }
+    userWalletBalance: {
+      btcWalletBalance: number
+      usdWalletBalance: number
+    }
+  } | null
+}> => {
+  // TODO move this logic to client side by setting up apollo clint
+  const session = await getServerSession(authOptions)
+  const token = session?.accessToken
+  const me = session?.userData.data
+  if (!me || !token) {
+    return {
+      error: true,
+      message: "User Not Found",
+      responsePayload: null,
+    }
+  }
+
+  const btcWallet = getBTCWallet(session.userData.data)
+  const usdWallet = getUSDWallet(session.userData.data)
+
+  if (!btcWallet || !usdWallet) {
+    return {
+      error: true,
+      message: "Wallet details not Found",
+      responsePayload: null,
+    }
+  }
+
+  const realtimePrice = await getRealtimePriceQuery(token)
+  if (realtimePrice instanceof Error) {
+    return {
+      error: true,
+      message: realtimePrice.message,
+      responsePayload: null,
+    }
+  }
+
+  const totalAmountForBTCWallet =
+    TotalAmount.wallets.BTC.SATS +
+    convertUsdToBtcSats(TotalAmount.wallets.USD, realtimePrice.data)
+  const totalAmountForUSDWallet = TotalAmount.wallets.USD * 100
+  if (
+    btcWallet?.balance >= totalAmountForBTCWallet &&
+    usdWallet?.balance >= totalAmountForUSDWallet
+  ) {
+    return {
+      error: false,
+      message: "success",
+      responsePayload: {
+        transactionAmount: {
+          totalAmountForBTCWallet,
+          totalAmountForUSDWallet,
+        },
+        userWalletBalance: {
+          btcWalletBalance: btcWallet?.balance,
+          usdWalletBalance: usdWallet?.balance,
+        },
+      },
+    }
+  }
+
+  return {
+    error: true,
+    message: "Insufficient Balance",
+    responsePayload: null,
+  }
```

Comment:
to be consistent, we should have the pattern that path with errors are before the success path